### PR TITLE
Use continuous batching by default

### PR DESCRIPTION
**Pull request by Wovchena**

### CODE REVIEW

File: src/cpp/src/llm_pipeline.cpp [hunk 1]
```diff
@@ -527,12 +528,24 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (plugin_config.find(ov::genai::scheduler_config.name()) != plugin_config.end()) {
         auto config_without_scheduler_config = plugin_config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 8;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:541]:
```suggestion
        scheduler_config.num_kv_blocks = 16;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 2]
```diff
@@ -547,12 +560,23 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 8;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:573]:
```suggestion
        scheduler_config.num_kv_blocks= 16;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 3]
```diff
@@ -547,12 +560,23 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.num_kv_blocks= 16;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:573]:
```suggestion
        scheduler_config.num_kv_blocks= 16;
        scheduler_config.enable_prefix_caching = true;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 4]
```diff
@@ -527,12 +528,24 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (plugin_config.find(ov::genai::scheduler_config.name()) != plugin_config.end()) {
         auto config_without_scheduler_config = plugin_config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.num_kv_blocks = 16;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:541]:
```suggestion
        scheduler_config.num_kv_blocks = 16;
        scheduler_config.enable_prefix_caching = true;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 5]
```diff
@@ -527,12 +528,25 @@ ov::genai::LLMPipeline::LLMPipeline(
...
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:541]:
```suggestion
        scheduler_config.num_kv_blocks = 32;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 6]
```diff
@@ -547,12 +561,24 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.num_kv_blocks= 16;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:574]:
```suggestion
        scheduler_config.num_kv_blocks= 32;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 7]
```diff
@@ -547,12 +561,24 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.num_kv_blocks= 32;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:574]:
```suggestion
        scheduler_config.num_kv_blocks= 64;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 8]
```diff
@@ -527,12 +528,25 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (plugin_config.find(ov::genai::scheduler_config.name()) != plugin_config.end()) {
         auto config_without_scheduler_config = plugin_config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.num_kv_blocks = 32;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:541]:
```suggestion
        scheduler_config.num_kv_blocks = 64;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 9]
```diff
@@ -547,12 +561,24 @@ ov::genai::LLMPipeline::LLMPipeline(
     const std::string& device,
     const ov::AnyMap& config
 ){
+    std::cout << "Using continuous batching backend.\n";
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:564]:
```suggestion
    // std::cout << "Using continuous batching backend.\n";
```

File: src/cpp/src/llm_pipeline.cpp [hunk 10]
```diff
@@ -547,12 +561,24 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        std::cout << "Found custom SchedulerConfig.\n";
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:571]:
```suggestion
        // std::cout << "Found custom SchedulerConfig.\n";
```

File: src/cpp/src/llm_pipeline.cpp [hunk 11]
```diff
@@ -527,12 +528,25 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (plugin_config.find(ov::genai::scheduler_config.name()) != plugin_config.end()) {
         auto config_without_scheduler_config = plugin_config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.num_kv_blocks = 64;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:541]:
```suggestion
        scheduler_config.cache_size = 1;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 12]
```diff
@@ -547,12 +561,24 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     auto start_time = std::chrono::steady_clock::now();
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.num_kv_blocks= 64;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:574]:
```suggestion
        scheduler_config.cache_size = 1;
```

**Review by andrei-kochin — CHANGES_REQUESTED** [review]

File: src/cpp/src/llm_pipeline.cpp [hunk 13]
```diff
@@ -531,12 +533,25 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     if (plugin_config.find(ov::genai::scheduler_config.name()) != plugin_config.end()) {
         auto config_without_scheduler_config = plugin_config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = true;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:547]:
```suggestion
        scheduler_config.enable_prefix_caching = true;
        scheduler_config.max_num_batched_tokens = 2048;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 14]
```diff
@@ -551,12 +566,24 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = true;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:580]:
```suggestion
        scheduler_config.enable_prefix_caching = true;
        scheduler_config.max_num_batched_tokens = 2048;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 15]
```diff
@@ -531,12 +533,26 @@ ov::genai::LLMPipeline::LLMPipeline(
...
         auto config_without_scheduler_config = plugin_config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = true;
+        scheduler_config.max_num_batched_tokens = 2048;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:548]:
```suggestion
        scheduler_config.max_num_batched_tokens = 2048;
        scheduler_config.dynamic_split_fuse = false;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 16]
```diff
@@ -551,12 +567,25 @@ ov::genai::LLMPipeline::LLMPipeline(
...
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = true;
+        scheduler_config.max_num_batched_tokens = 2048;
```
Comment (andrei-kochin) [inline src/cpp/src/llm_pipeline.cpp:582]:
```suggestion
        scheduler_config.max_num_batched_tokens = 2048;
        scheduler_config.dynamic_split_fuse = false;
```

File: src/cpp/src/continuous_batching_impl.cpp [hunk 17]
```diff
@@ -269,6 +273,16 @@ ContinuousBatchingPipeline::ContinuousBatchingImpl::generate(const std::vector<o
...
     }
 
+    if (!continue_generation && !m_requests.empty()) {
+        SequenceGroup::Ptr request = m_requests[0];
+        for (const auto& sequence: request->get_sequences()) {
+            if (m_scheduler->has_block_table(sequence->get_id())) {
+                m_scheduler->free_sequence(sequence->get_id());
+            }
+        }
+        m_sampler->clear_beam_search_info(request->get_request_id());
```
Comment (ilya-lavrenov) [inline src/cpp/src/continuous_batching_impl.cpp:283]:
```suggestion
        m_sampler->clear_beam_search_info(request->get_request_id());
        m_requests.erase(m_requests.begin());
```

File: src/cpp/src/llm_pipeline.cpp [hunk 18]
```diff
@@ -554,12 +571,26 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = true;
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:585]:
```suggestion
        scheduler_config.enable_prefix_caching = false;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 19]
```diff
@@ -534,12 +536,27 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     if (plugin_config.find(ov::genai::scheduler_config.name()) != plugin_config.end()) {
         auto config_without_scheduler_config = plugin_config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = true;
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:550]:
```suggestion
        scheduler_config.enable_prefix_caching = false;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 20]
```diff
@@ -534,12 +536,27 @@ ov::genai::LLMPipeline::LLMPipeline(
...
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = false;
+        scheduler_config.max_num_batched_tokens = 2048;
+        scheduler_config.dynamic_split_fuse = false;
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:552]:
```suggestion
        scheduler_config.dynamic_split_fuse = true;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 21]
```diff
@@ -554,12 +571,26 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = false;
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:585]:
```suggestion
```

File: src/cpp/src/llm_pipeline.cpp [hunk 22]
```diff
@@ -554,12 +571,26 @@ ov::genai::LLMPipeline::LLMPipeline(
...
+        scheduler_config.max_num_batched_tokens = 2048;
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:586]:
```suggestion
        scheduler_config.max_num_batched_tokens = 256;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 23]
```diff
@@ -534,12 +536,27 @@ ov::genai::LLMPipeline::LLMPipeline(
...
         auto config_without_scheduler_config = plugin_config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = false;
+        scheduler_config.max_num_batched_tokens = 2048;
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:551]:
```suggestion
```

File: src/cpp/src/llm_pipeline.cpp [hunk 24]
```diff
@@ -534,12 +536,26 @@ ov::genai::LLMPipeline::LLMPipeline(
...
         auto config_without_scheduler_config = plugin_config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = plugin_config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = false;
+        scheduler_config.dynamic_split_fuse = true;
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:551]:
```suggestion
        scheduler_config.dynamic_split_fuse = true;
```

File: src/cpp/src/llm_pipeline.cpp [hunk 25]
```diff
@@ -554,12 +570,25 @@ ov::genai::LLMPipeline::LLMPipeline(
...
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
         auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(path, scheduler_config, device, config_without_scheduler_config);
+        // std::cout << "Found custom SchedulerConfig.\n";
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.max_num_batched_tokens = 256;
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:584]:
```suggestion
        scheduler_config.enable_prefix_caching = false;
```

File: .github/workflows/causal_lm_cpp.yml [hunk 26]
```diff
@@ -479,7 +479,7 @@ jobs:
               predicted_greedy = f.readline()
           with open('predictions_prompt_lookup.txt', 'r') as f:
               predicted_prompt_lookup = f.readline()
-          assert predicted_greedy == predicted_prompt_lookup
+          assert predicted_greedy == predicted_prompt_lookup, "Expected {predicted_greedy}, actual {predicted_prompt_lookup}"
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:482]:
```suggestion
          assert predicted_greedy == predicted_prompt_lookup, 'Expected {predicted_greedy}, actual {predicted_prompt_lookup}'
```

File: .github/workflows/causal_lm_cpp.yml [hunk 27]
```diff
@@ -501,7 +501,7 @@ jobs:
...
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:504]:
```suggestion
          assert predicted_greedy == predicted_prompt_lookup, 'Expected {predicted_greedy}, actual {predicted_prompt_lookup}'
```

File: .github/workflows/causal_lm_cpp.yml [hunk 28]
```diff
@@ -479,7 +479,7 @@ jobs:
               predicted_greedy = f.readline()
           with open('predictions_prompt_lookup.txt', 'r') as f:
               predicted_prompt_lookup = f.readline()
-          assert predicted_greedy == predicted_prompt_lookup
+          assert predicted_greedy == predicted_prompt_lookup, 'Expected {predicted_greedy}, actual {predicted_prompt_lookup}'
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:482]:
```suggestion
          assert predicted_greedy == predicted_prompt_lookup, f'Expected {predicted_greedy}, actual {predicted_prompt_lookup}'
```

File: .github/workflows/causal_lm_cpp.yml [hunk 29]
```diff
@@ -501,7 +501,7 @@ jobs:
...
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:504]:
```suggestion
          assert predicted_greedy == predicted_prompt_lookup, f'Expected {predicted_greedy}, actual {predicted_prompt_lookup}'
```

File: .github/workflows/causal_lm_cpp.yml [hunk 30]
```diff
@@ -285,7 +285,7 @@ jobs:
       - run: >
           . ./ov/setupvars.sh
           && export PYTHONPATH=./build/:$PYTHONPATH
-          && timeout 2m ./build/samples/cpp/greedy_causal_lm/greedy_causal_lm ./Qwen-7B-Chat/ 69 | diff <(timeout 2m samples/python/greedy_causal_lm/greedy_causal_lm.py ./Qwen-7B-Chat/ 69) -
+          && timeout 2m ./build/samples/cpp/greedy_causal_lm/greedy_causal_lm ./Qwen-7B-Chat/ 69 | diff <(timeout 4m samples/python/greedy_causal_lm/greedy_causal_lm.py ./Qwen-7B-Chat/ 69) -
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:288]:
```suggestion
          && timeout 2m ./build/samples/cpp/greedy_causal_lm/greedy_causal_lm ./Qwen-7B-Chat/ 69 | diff <(timeout 2m samples/python/greedy_causal_lm/greedy_causal_lm.py ./Qwen-7B-Chat/ 69) -
```

File: .github/workflows/causal_lm_cpp.yml [hunk 31]
```diff
@@ -708,7 +708,7 @@ jobs:
       - name: Run visual_language_chat sample - MiniCPM-V-2_6
         run: >
           source ./ov/setupvars.sh
-          && timeout 120s ./build/samples/cpp/visual_language_chat/visual_language_chat ./miniCPM-V-2_6/ cat.jpg
+          && timeout 240s ./build/samples/cpp/visual_language_chat/visual_language_chat ./miniCPM-V-2_6/ cat.jpg
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:711]:
```suggestion
          && timeout 120s ./build/samples/cpp/visual_language_chat/visual_language_chat ./miniCPM-V-2_6/ cat.jpg
```

File: .github/workflows/causal_lm_cpp.yml [hunk 32]
```diff
@@ -729,7 +729,7 @@ jobs:
           source ./ov/setupvars.sh
           export PYTHONPATH=./build/:$PYTHONPATH
           printf 'What is on the image?\nWhat is special on the image?\n' > ./input.txt
-          timeout 120s python ./samples/python/visual_language_chat/visual_language_chat.py ./miniCPM-V-2_6/ cat.jpg < input.txt > ./pred.txt
+          timeout 240s python ./samples/python/visual_language_chat/visual_language_chat.py ./miniCPM-V-2_6/ cat.jpg < input.txt > ./pred.txt
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:732]:
```suggestion
          timeout 120s python ./samples/python/visual_language_chat/visual_language_chat.py ./miniCPM-V-2_6/ cat.jpg < input.txt > ./pred.txt
```

File: src/python/py_generate_pipeline.cpp [hunk 33]
```diff
@@ -422,6 +424,7 @@ PYBIND11_MODULE(py_generate_pipeline, m) {
             ScopedVar env_manager(utils::ov_tokenizers_module_path());
             return std::make_unique<LLMPipeline>(model_path, tokenizer, device, utils::properties_to_any_map(config));
         }),
+        py::call_guard<py::scoped_ostream_redirect, py::scoped_estream_redirect>(),  // Respect std::cout flushes from constructor.
```
Comment (ilya-lavrenov) [inline src/python/py_generate_pipeline.cpp:427]: does it need to be ported on master ?

Comment (Wovchena) [reply src/python/py_generate_pipeline.cpp:427]: It was added to make constructor prints visible at the correct location. But the prints are commented out now and are going to be removed completely before the merge. That means it shouldn't be ported to master in that state.

File: src/python/py_generate_pipeline.cpp [hunk 34]
```diff
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <filesystem>
+#include <pybind11/iostream.h>
```
Comment (ilya-lavrenov) [inline src/python/py_generate_pipeline.cpp:5]:
```suggestion
```

File: src/python/py_generate_pipeline.cpp [hunk 35]
```diff
@@ -403,6 +404,7 @@ PYBIND11_MODULE(py_generate_pipeline, m) {
             ScopedVar env_manager(utils::ov_tokenizers_module_path());
             return std::make_unique<LLMPipeline>(model_path, device, utils::properties_to_any_map(config));
         }),
+        py::call_guard<py::scoped_ostream_redirect, py::scoped_estream_redirect>(),  // Respect std::cout flushes from constructor.
```
Comment (ilya-lavrenov) [inline src/python/py_generate_pipeline.cpp:407]:
```suggestion
```

File: src/python/py_generate_pipeline.cpp (hunk 33 above)
Comment (ilya-lavrenov) [inline src/python/py_generate_pipeline.cpp:427]:
```suggestion
```

File: src/cpp/src/llm_pipeline.cpp [hunk 36]
```diff
@@ -570,12 +585,23 @@ ov::genai::LLMPipeline::LLMPipeline(
     const std::string& device,
     const ov::AnyMap& config
 ){
+    // std::cout << "Using continuous batching backend.\n";
     auto start_time = std::chrono::steady_clock::now();
     if (config.find(ov::genai::scheduler_config.name()) != config.end()) {
         auto config_without_scheduler_config = config;
         config_without_scheduler_config.erase(ov::genai::scheduler_config.name());
-        auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
+        auto& scheduler_config = properties.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:593]:
```suggestion
        auto& scheduler_config = config.at(ov::genai::scheduler_config.name()).as<SchedulerConfig>();
```

File: src/cpp/src/llm_pipeline.cpp [hunk 37]
```diff
@@ -570,12 +585,23 @@ ov::genai::LLMPipeline::LLMPipeline(
...
         m_pimpl = std::make_unique<ContinuousBatchingAdapter>(models_path, scheduler_config, device, config_without_scheduler_config);
+    } else if (true) {
+        SchedulerConfig scheduler_config;
+        scheduler_config.cache_size = 1;
+        scheduler_config.enable_prefix_caching = false;
+        m_pimpl = std::make_unique<ContinuousBatchingAdapter>(
+            models_path,
+            scheduler_config,
+            device,
+            properties
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:603]:
```suggestion
            config
```

File: src/cpp/src/llm_pipeline.cpp [hunk 38]
```diff
@@ -623,7 +645,7 @@ void ov::genai::LLMPipeline::set_generation_config(const GenerationConfig& confi
     // if eos_token_id was not provided in config forward from default config
     if (config.eos_token_id == -1)
         m_pimpl->m_generation_config.set_eos_token_id(default_eos_token_id);
-
+
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:648]:
```suggestion
```

**Comment by ilya-lavrenov** [comment]
for some reason only a single test is run:
```
======================== 1 failed, 3 warnings in 45.75s ========================
```
@Wovchena any ideas?

**Comment by Wovchena** [comment]
It was meant to run a single test only: the command is `python -m pytest -v ./tests/python_tests/test_chat_generate_api.py::test_set_chat_template`. The full testing is in the next step.

**Comment by ilya-lavrenov** [comment]
Replaced by https://github.com/openvinotoolkit/openvino.genai/pull/1455

**Pull request closed without merging**